### PR TITLE
 parser: support more analyze options

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

Support more analyze options.
### What is changed and how it works?

Add new syntax like `analyze table t with 4 buckets, 4 topn, 4 cmsketch width, 4 cmsketch depth`.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - None

Related changes

 - Need to update the documentation

For https://github.com/pingcap/tidb/pull/11278

### CODE REVIEW

File: ast/stats.go [hunk 1]
```diff
@@ -32,13 +32,38 @@ type AnalyzeTableStmt struct {
...
+// Analyze option types.
+const (
+	AnalyzeOptNumBuckets = iota
+	AnalyzeOptNumTopN
+	AnalyzeOptCMSketchDepth
+	AnalyzeOptCMSketchWidth
+)
+
+// AnalyzeOptionString stores the string form of analyze options.
+var AnalyzeOptionString = map[AnalyzeOptionType]string{
```
Comment (winoros) [inline ast/stats.go]: We can use a slice directly?

**Comment by lamxTyler** [comment]
@lzmhhh123 It does not matter if the internal implementation changed as long as the syntax is the same.

File: ast/stats.go (hunk 1 above)
Comment (lamxTyler) [reply ast/stats.go]: Yes, but `map` could reduce our mental burden by maintaining their relationships explicitly.

**Comment by eurekaka** [comment]
@lamxTyler CI failed, PTAL

**Comment by lamxTyler** [comment]
@eurekaka It is expected since it removed some asts, that is it requires the tidb version of https://github.com/pingcap/tidb/pull/11278.

**Pull request merged**